src/api/libcellml/issue.h

```diff
@@ -0,0 +1,433 @@
+     *
+     * Return @c true if the @p cause matches the the @c Issue::Cause for this
+     * issue and @c false otherwise.
+     *
+     * @return @c true if the @p cause matches the the @c Issue::Cause for this
+     * issue and @c false otherwise.
+     */
+    bool isCause(Cause cause) const;
+
+    /**
+     * @brief Set the level of this issue.
+     *
+     * Set the @p level of this issue from the options available in
+     * @c Issue::Level.
+     *
+     * @param level The @c Issue::Level to set.
+     */
+    void setLevel(Level level);
+
+    /**
+     * @brief Get the level of this issue.
+     *
+     * Get the @c level of this issue. If no level has been set for
+     * this issue, will return Level::ERROR.
+     *
+     * @return The @c Issue::Level set for this issue.
+     */
+    Level level() const;
+
+    /**
+     * @brief Check whether the level of this issue matches the argument level.
+     *
+     * Return @c true if the @p level matches the the @c Issue::Level for this
+     * issue and @c false otherwise.
+     *
+     * @return @c true if the @p level matches the the @c Issue::Level for this
+     * issue and @c false otherwise.
+     */
+    bool isLevel(Level level) const;
+
+    /**
+     * @brief Set the @c enum ReferenceRule of this issue.
+     *
+     * Set the @p rule of this issue from the options available in
+     * the @c ReferenceRule @c enum.
+     *
+     * @param rule The @c ReferenceRule to set.
+     */
+    void setRule(ReferenceRule rule);
+
+    /**
+     * @brief Get the @c enum ReferenceRule of this issue.
+     *
+     * Get the @c enum @c ReferenceRule value @p rule of this issue. If
+     * no rule has been set for this issue, will return ReferenceRule::UNDEFINED.
+     *
+     * @return The @c ReferenceRule for this issue.
+     */
+    ReferenceRule rule() const;
+
+    /**
+     * @brief Set the @c url of this issue.
+     *
+     * Manually set the @c url to which the user should refer for more information on the
+     * specification rule which has been broken or guidelines that haven't been followed.
+     * If this has not been set manually by use of this function, the default root url
+     * will be used (specified in baseIssueUrl) in conjunction with the referenceHeading
+     * string.
+     *
+     * @param rule The @c url to set manually.
+     */
+    void setUrl(std::string &url) const;
+
+    /**
+     * @brief Get the @c url of this issue.
+     *
+     * Get the @c url to which the user should refer for more information on this issue.
+     *
+     * @return The @c url for this issue.
+     */
+    std::string url() const;
+
+    /**
+     * @brief Get the @c std::string heading associated with the @c enum ReferenceRule for this issue.
+     *
+     * Get the @c std::string CellML 2.0 Specification heading associated with the @c enum ReferenceRule
+     * for this issue. If no rule has been set for this issue, will return an empty string.
```

Comment:
for non-specification issues?

Reply:
You're right.  I'll change it to "CellML 2.0 Reference heading" (unless there's something better to call it?)

Reply:
Done.